### ISSUE TITLE
Flutter web return 404 on windows when Flutter SDK not installed under C:\

### ISSUE THREAD

**Issue by ctrysbita** [flutter/flutter#51841]
## Steps to Reproduce

1. Install Flutter SDK under `D:\Programs\flutter`
2. Create an example project by `flutter create example`
3. `cd example ; flutter run -d chrome`

![QQ截图20200303194349](https://user-images.githubusercontent.com/8357481/75772735-7732af80-5d87-11ea-84f6-9a5a6dd03f30.png)

## Logs

```
PS C:\Users\example\Documents\web> flutter run -d chrome --verbose
[  +15 ms] executing: [D:\Programs\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +42 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 67826bdce54505760fe83b7ead70bdb5af6fe9f2
[        ] executing: [D:\Programs\flutter/] git describe --match v*.*.* --first-parent --long --tags
[  +24 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[  +10 ms] v1.15.3-0-g67826bdce
[   +5 ms] executing: [D:\Programs\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +31 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/dev
[        ] executing: [D:\Programs\flutter/] git ls-remote --get-url origin
[  +21 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter
[  +46 ms] executing: [D:\Programs\flutter/] git rev-parse --abbrev-ref HEAD
[  +23 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] dev
[  +30 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +130 ms] Unable to locate an Android SDK.
[ +116 ms] Unable to locate an Android SDK.
[  +64 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +8 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +112 ms] Generating C:\Users\example\Documents\web\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +75 ms] Launching lib\main.dart on Chrome in debug mode...
[  +18 ms] Updating assets
[  +67 ms] Syncing files to device Chrome...
[  +28 ms] Generating C:\Users\example\Documents\web\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[ +104 ms] <- reset
[   +4 ms] D:\Programs\flutter\bin\cache\dart-sdk\bin\dart.exe D:\Programs\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root
D:\Programs\flutter\bin\cache\flutter_web_sdk/ --incremental --target=dartdevc -Ddart.developer.causal_async_stacks=true --output-dill
C:\Users\example\AppData\Local\Temp\flutter_tool.18043e25-5d44-11ea-b52c-a4c3f08e0e53\app.dill --packages C:\Users\example\Documents\web\.packages -Ddart.vm.profile=false 
-Ddart.vm.product=false --bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions    
--enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root --platform file:///D:/Programs/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill    
[  +17 ms] <- compile C:\Users\example\AppData\Local\Temp\flutter_tools.18043e26-5d44-11ea-b52c-a4c3f08e0e53\web_entrypoint.dart
[+8467 ms] Syncing files to device Chrome... (completed in 8,617ms, longer than expected)
[   +1 ms] Synced 20.9MB.
[        ] <- accept     
[ +236 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[        ] been thoroughly tested in production environments.
[        ] For more information see https://flutter.dev/web
[        ]   To hot restart changes while running, press "r". To hot restart (and refresh the browser), press "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
```

## Doctor Summary

```
[√] Flutter (Channel dev, v1.15.3, on Microsoft Windows [Version 10.0.18363.657], locale zh-CN)
    • Flutter version 1.15.3 at D:\Programs\flutter
    • Framework revision 67826bdce5 (3 weeks ago), 2020-02-10 14:59:32 -0800
    • Engine revision 6158f03ef5
    • Dart version 2.8.0 (build 2.8.0-dev.8.0 514a8d4c84)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_SDK_ROOT to that location.
      You may also want to add it to your PATH environment variable.


[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).

[√] VS Code, 64-bit edition (version 1.42.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 79.0.3945.88
    • Web Server • web-server • web-javascript • Flutter Tools
```

**Comment by jonahwilliams** [comment]
Would be worth trying on the latest dev, I think its fixed but I only have a `C:/` at home 🕵

**Comment by ctrysbita** [comment]
Still get 404 while using the latest dev release. (same as before)

## Doctor summary
```
[√] Flutter (Channel dev, v1.15.14, on Microsoft Windows [Version 10.0.18363.657], locale zh-CN)
    • Flutter version 1.15.14 at D:\Programs\flutter
    • Framework revision a68f96ae3f (3 hours ago), 2020-03-05 16:08:44 -0800
    • Engine revision e28d26e305
    • Dart version 2.8.0 (build 2.8.0-dev.12.0 9983424a3c)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_SDK_ROOT to that location.
      You may also want to add it to your PATH environment variable.


[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).

[√] VS Code, 64-bit edition (version 1.42.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 79.0.3945.88
    • Web Server • web-server • web-javascript • Flutter Tools
```

**Comment by ctrysbita** [comment]
Fixed in v1.15.20 dev.
